api/api.go

```diff
@@ -176,8 +176,8 @@ func (a *API) ListenExecution(service string, f *ExecutionFilter) (*ExecutionLis
 }
 
 // SubmitResult submits results for executionID.
-func (a *API) SubmitResult(executionID string, outputKey string, outputs []byte) error {
```

Comment:
Maybe I'm missing some parts here but `SubmitResult` should have both the outputs and errors.
The execution already has an error field so why not getting it from `SubmitResultRequest_Error`. Here when there is an error we store it as valid data which should actually break the validation in the execution package `RequireInputs` and we ignore this data.

But maybe I'm missing something. I haven't tried it yet but I think there is an error with the validation of the data when we save the result. https://github.com/mesg-foundation/engine/blob/625840c3879812e62a732b38ab1945a6263efac1/execution/execution.go#L60